code/datums/components/_component.dm

```diff
@@ -310,18 +312,16 @@
 /datum/proc/_SendSignal(sigtype, list/arguments)
 	var/target = comp_lookup[sigtype]
 	if(!length(target))
-		var/datum/C = target
-		if(!(C.datum_flags & DF_SIGNAL_ENABLED))
+		var/datum/listening_datum = target
+		if(!(listening_datum.datum_flags & DF_SIGNAL_ENABLED))
 			return NONE
-		var/proctype = C.signal_procs[src][sigtype]
-		return NONE | CallAsync(C, proctype, arguments)
+		return NONE | CallAsync(listening_datum, listening_datum.signal_procs[src][sigtype], arguments)
 	. = NONE
-	for(var/I in target)
-		var/datum/C = I
-		if(!(C.datum_flags & DF_SIGNAL_ENABLED))
+	for(var/datum/listening_datum as anything in target)
+		if(!(listening_datum.datum_flags & DF_SIGNAL_ENABLED))
 			continue
-		var/proctype = C.signal_procs[src][sigtype]
-		. |= CallAsync(C, proctype, arguments)
+		var/proctype = listening_datum.signal_procs[src][sigtype]
+		. |= CallAsync(listening_datum, proctype, arguments)
```

Comment:
You may as well do the same thing here as above

Reply:
whoops missed that

Reply:
done